truffle/src/main/ruby/core/complex.rb

```diff
@@ -53,9 +53,10 @@ class Complex < Numeric
   undef_method :i
 
   def self.convert(real, imag = undefined)
-    if real.equal?(nil) || imag.equal?(nil)
+    if nil.equal?(real) || nil.equal?(imag)
```

Comment:
they might be `undefined`, and calling methods on `undefined` does not work, on purpose.